src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaImplementationTests.cs

```diff
@@ -61,7 +122,481 @@ public static void UseAfterDispose()
+                    AssertExtensions.Throws<ArgumentException>("source", () => MLDsa.ImportFromPem(pem.AsSpan())));
+            }
+        }
+
+        [Fact]
+        public static void ImportFromEncryptedPem_MalformedPem()
+        {
+            AssertThrows(WritePemRaw("UNKNOWN LABEL", []));
+            AssertThrows(WritePemRaw("CERTIFICATE", []));
+            AssertThrows(string.Empty);
+            AssertThrows(WritePemRaw("ENCRYPTED PRIVATE KEY", []) + '\n' + WritePemRaw("ENCRYPTED PRIVATE KEY", []));
+            AssertThrows(WritePemRaw("ENCRYPTED PRIVATE KEY", "%"));
+
+            static void AssertThrows(string encryptedPem)
+            {
+                AssertThrowIfNotSupported(() =>
+                    AssertExtensions.Throws<ArgumentException>("source", () => MLDsa.ImportFromEncryptedPem(encryptedPem, "PLACEHOLDER")));
+                AssertThrowIfNotSupported(() =>
+                    AssertExtensions.Throws<ArgumentException>("source", () => MLDsa.ImportFromEncryptedPem(encryptedPem, "PLACEHOLDER"u8)));
+                AssertThrowIfNotSupported(() =>
+                    AssertExtensions.Throws<ArgumentException>("source", () => MLDsa.ImportFromEncryptedPem(encryptedPem.AsSpan(), "PLACEHOLDER")));
+                AssertThrowIfNotSupported(() =>
+                    AssertExtensions.Throws<ArgumentException>("source", () => MLDsa.ImportFromEncryptedPem(encryptedPem, "PLACEHOLDER"u8.ToArray())));
+            }
+        }
+
+        [ConditionalTheory(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+        [MemberData(nameof(MLDsaTestsData.AllMLDsaAlgorithms), MemberType = typeof(MLDsaTestsData))]
+        public static void AlgorithmMatches_GenerateKey(MLDsaAlgorithm algorithm)
+        {
+            byte[] publicKey = new byte[algorithm.PublicKeySizeInBytes];
+            byte[] secretKey = new byte[algorithm.SecretKeySizeInBytes];
+            byte[] privateSeed = new byte[algorithm.PrivateSeedSizeInBytes];
+            AssertThrowIfNotSupported(() =>
+            {
+                using MLDsa mldsa = MLDsa.GenerateKey(algorithm);
+                mldsa.ExportMLDsaPublicKey(publicKey);
+                mldsa.ExportMLDsaSecretKey(secretKey);
+                mldsa.ExportMLDsaPrivateSeed(privateSeed);
+                Assert.Equal(algorithm, mldsa.Algorithm);
+            });
+
+            MLDsaTestHelpers.AssertImportPublicKey(import =>
+                AssertThrowIfNotSupported(() =>
+                    Assert.Equal(algorithm, import().Algorithm)), algorithm, publicKey);
+
+            MLDsaTestHelpers.AssertImportSecretKey(import =>
+                AssertThrowIfNotSupported(() =>
+                    Assert.Equal(algorithm, import().Algorithm)), algorithm, secretKey);
+
+            MLDsaTestHelpers.AssertImportPrivateSeed(import =>
+                AssertThrowIfNotSupported(() =>
+                    Assert.Equal(algorithm, import().Algorithm)), algorithm, privateSeed);
```

Comment:
Neither these tests (or other successful import tests) nor the helper (because it can't see the return) ever dispose these keys.

They're not really side-effecting (unlike PFX import on Windows), so it's not a _huge_ problem... but if there's an easy way to get these to dispose... that would be goodness.